### PR TITLE
Spring Mean Piston Position

### PR DESCRIPTION
**Pull request by andermi**
may now specify desired mean piston position in empy

to test, in a sourced terminal:

```
$ empy -D 'inc_wave_spectrum_type="None"' -o install/buoy_description/share/buoy_description/models/mbari_wec/model.sdf install/buoy_description/share/buoy_description/models/mbari_wec/model.sdf.em
$ ros2 launch buoy_gazebo mbari_wec.launch.py extra_gz_args:='-rs' regenerate_models:=false
```

and in another sourced terminal:
```
$ ros2 topic echo /spring_data
```

and check that range_finder settles close to the desired default value of 0.7m

can also define other values:
```
$ empy -D 'x_mean_pos=0.9' -D 'inc_wave_spectrum_type="None"' -o install/buoy_description/share/buoy_description/models/mbari_wec/model.sdf install/buoy_description/share/buoy_description/models/mbari_wec/model.sdf.em
```

should settle close to 0.9m

Also added to batch parameters and model regen in tests

TODO:
- figure out what I did wrong in free-body forces (unexpected fudge factor required)

### CODE REVIEW

**Comment by hamilton8415** [comment]
Impressive job.  I will run and test on friday afternoon.

File: buoy_description/models/mbari_wec/model.sdf.em [hunk 1]
```diff
@@ -83,6 +83,71 @@ elif 'Custom' in inc_wave_spectrum_type:
...
+
+V0_u = 0.0431
+V0_l = 0.0574
+P0_u = 410152
+P0_l = 1212060
+T0_u = 263.15
+T0_l = 283.15
+
+m_u = P0_u*V0_u / (R_specific*T0_u)
+m_l = P0_l*V0_l / (R_specific*T0_l)
```
Comment (andermi) [inline buoy_description/models/mbari_wec/model.sdf.em:137]: I probably need to comment all these variables and the point of it all

File: buoy_description/models/mbari_wec/model.sdf.em [hunk 2]
```diff
@@ -83,6 +83,71 @@ elif 'Custom' in inc_wave_spectrum_type:
...
+m_delta = spo.fsolve(lambda m_delta: x_mean_pos - x_eq(Ap_u, Ap_l,
+                                                       m_u - m_delta,
+                                                       m_l + m_delta,
+                                                       Vd_u, Vd_l, c), m_delta_guess)[0]
+V0_u = Ap_u*x_mean_pos + Vd_u
+V0_l = Ap_l*(2.03 - x_mean_pos) + Vd_l
+m_u = m_u - m_delta
+m_l = m_l + m_delta
+P0_u = m_u*R_specific*T0_u / V0_u
+P0_l = m_l*R_specific*T0_l / V0_l
```
Comment (andermi) [inline buoy_description/models/mbari_wec/model.sdf.em:150]: I probably need to comment all these variables and the point of it all

**Comment by hamilton8415** [comment]
Running nicely for me.  As for the tests, I will take to improve the hard-coded paths.  For the experiment comparison, as you suggest seems fine.  Choices seem to be to have the experiment comparison not match in pressure, or have the default piston position be pretty low.  mean position as a launch file argument that would allow the experiment test  to set it where it needs to?  I think that's a bigger discussion about how/where all of these burgeoning parameters can be set from...  Monday.

**Comment by andermi** [comment]
> mean position as a launch file argument that would allow the experiment test to set it where it needs to?

Yeah we could definitely do that -- good idea!

**Comment by andermi** [comment]
should be all set!

**Comment by andermi** [comment]
@hamilton8415 Go ahead and merge if you'd like to

**Review by hamilton8415 — APPROVED** [review]
Nice Job.  Runs as expected for me.  I will merge into fs_hydrodynamics and continue checking behaviors there.

**Pull request merged**